src/main/java/org/zalando/nakadi/security/Client.java

```diff
@@ -1,22 +1,29 @@
 package org.zalando.nakadi.security;
 
-public interface Client {
+import org.zalando.nakadi.exceptions.IllegalScopeException;
 
-    Client PERMIT_ALL = clientId -> true;
+import java.util.Set;
 
-    boolean is(String clientId);
+public class Client implements IClient {
```

Comment:
I see everybody against it :) I know it is a bad practice to use such names for the interfaces but I thought it would straight forward here. Renaming happened because I think name AuthorizedClient is not clear here. This class is used to check clientId, in particular that client who is requesting resource is the client who is owning this resources, that implies that AuthorizedClient does authorization and the client is not authorized until we do AuthorizedClient.is(clientId). I could not find something better than IClient.